### ISSUE TITLE
Stamps on top and bottom surfaces don't have fill behind them

### ISSUE THREAD

**Issue by rcmaniac25** [MosaicManufacturing/canvas-issues#140]
**Describe the bug**
First off, I see this as a bug, not a feature. Given that stamps are a picture projected onto a surface, I'd assume the surface is the background and that the colors making up the image will be placed on them. What seems to happen is the image simply replaces the surface/background. In my case, the print preview seems to indicate I will be able to see the infill through the stamped image. I'd expect that the surface will be used as fill over/under the stamp so it has a clean surface to adhere to and won't be visible through the stamp. Note: I know the preview isn't perfect. I haven't tried printing yet.

**To Reproduce**
1. Add a model that has a flat top or bottom.
2. Add a stamp to the top or bottom surface.
3. Slice

**Expected behavior**
Best-effort stamp applied to a surface, without infill visible.

**Screenshots**
Expectation (with my amazing /s photoshopping skills)
![t3](https://user-images.githubusercontent.com/418272/75513074-974a2380-59c1-11ea-81c9-5a8b75e892dd.jpg)
Reality
![t3](https://user-images.githubusercontent.com/418272/75512926-fce9e000-59c0-11ea-860c-ec40b7b9d2f5.jpg)
Gaps with infill visible (not everything circled)
![t3](https://user-images.githubusercontent.com/418272/75513179-0758a980-59c2-11ea-9811-608afb294596.jpg)

**Operating System (please complete the following information):**
 - OS: Windows
 - Browser: Firefox
 - Version: 72

**Additional context**

**Comment by brandonbloch** [comment]
Hi @rcmaniac25,

You're correct, this is a bug. It's not actually caused by stamping, however, but rather by poor gap fill generation. With such fine details in stamped models, the issue just presents itself more noticeably than for non-stamped models. Know that we're working on improving the algorithm for gap fill to improve print quality, especially in situations like this.

Cheers,
Brandon

**Comment by rcmaniac25** [comment]
I agree that this isn't a bug of stamping, but stamping makes it really easy to make it occur without making a complex model.
